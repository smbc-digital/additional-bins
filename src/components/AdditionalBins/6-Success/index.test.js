import React from 'react'
import Enzyme, { mount } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import { Success } from './index'
import renderer from 'react-test-renderer'


Enzyme.configure({ adapter: new Adapter() })
describe('Success',() => { 

it('should find elements required',() => { 
    // Arrange
    const context = {
        success: {
            value: '',
        }
    }
    const wrapper = mount(<Success context={context} />)
    // Act

    // Assert
    expect(wrapper.find('h1','h2','p','a').exists()).toBe(true)
})

it('should show triage success page when theres no receipt number',() => { 
    // Arrange
    const context = {
        success: {
            value: '',
        },
        receiptNumber: ''
    }
    const wrapper = mount(<Success context={context} />)

    // Act

    // Assert
    expect(wrapper.find('a').text()).toEqual('Download a waste diary (PDF 66Kb)')
})

it('should show non triage success page when theres a receipt number',() => { 
    // Arrange
    const context = {
        success: {
            value: '',
        },
        receiptNumber: '1'
    }
    const wrapper = mount(<Success context={context} />)

    // Act

    // Assert
    expect(wrapper.find('h3').exists()).toBe(false)
})

    describe('snapshot', () => {
    it('renders correctly', () => {
        const context = {
            
            }


        const tree = renderer
            .create(< Success context={context} />)
            .toJSON()
        expect(tree).toMatchSnapshot()
    })
})


})