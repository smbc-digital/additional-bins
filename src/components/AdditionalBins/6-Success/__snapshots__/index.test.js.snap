// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Success snapshot renders correctly 1`] = `
Array [
  <section
    className="header-container"
  >
    <h1>
      Request an additional black bin
    </h1>
    <p
      className="h2"
    >
      Thanks for your request
    </p>
  </section>,
  <p>
    <strong>
      Your reference number is:
    </strong>
     
     
  </p>,
  <p>
    Please keep this number safe as you'll need it for the next step. You'll also receive an email with this information.
  </p>,
  <h2>
    What happens next
  </h2>,
  <p>
    You must download and fill in a waste diary to help us understand why you need an additional black bin. You'll need to record the waste you put in your black bin over 7 days.
  </p>,
  <p>
    You can use the button below to download the waste diary. You'll also receive a link in the email if you want to complete it later. The email will tell you how to return your waste diary to us once youâ€™ve completed it.
  </p>,
  <a
    className="button-primary"
    href="https://s3-eu-west-1.amazonaws.com/live-iag-static-assets/pdf/BinCalendars/Waste+diary+v2.pdf"
    rel="noopener noreferrer"
    target="_blank"
  >
    Download a waste diary (PDF 66Kb)
  </a>,
]
`;
